switools/signaturelib.py

```diff
@@ -12,4 +13,23 @@ def getSigFileName( swiFile ):
       return SWIX_SIG_FILE_NAME
    return SWI_SIG_FILE_NAME
 
+def getOptimizations( swi, workDir ):
+   ret = os.system( "set -e; swi=$(readlink -f %s); cd %s; "
+                    "unzip -q -o $swi swimSqshMap" % ( swi, workDir ) )
+   if ret:
+      return None # legacy image
+   optims = []
+   with open( "%s/swimSqshMap" % workDir ) as f:
+      for line in f:
+         optim, _ = line.split( "=", 1 )
+         optims.append( optim )
+   os.system( "rm %s/swimSqshMap" % workDir )
+   return optims
 
+def extractSwadapt( swi, workDir ):
+   ret = os.system( "set -e; image=$(readlink -f %s); cd %s;"
+                    "unzip -o -q $image swadapt" % ( swi, workDir ) )
+   if ret:
+      print( "Error: '%s' does not contain the 'swadapt' utility" % swi )
+      shutil.rmtree( workDir )
+      sys.exit( -1 )
```

Comment:
Will it always be safe to exit from this function? Maybe better to throw some exception?

Reply:
Changed it to do all cleanups up top.